langchain4j-core/src/main/java/dev/langchain4j/rag/content/Content.java

```diff
@@ -53,6 +62,7 @@ public int hashCode() {
     public String toString() {
         return "Content {" +
                 " textSegment = " + textSegment +
+                ", metadata = " + metadata +
```

Comment:
minor: worth leaving a comment in `equals()` and `hashCode()` that `metadata` does not participate in comparisons for {reasons}

Reply:
Or perhaps it should participate in the comparisons, but we need to reimplement comparison logic in `ReciprocalRankFuser`?

Reply:
I will merge this PR in its current state now to unblock a few users waiting for this feature, but we need to revisit this